### ISSUE TITLE
NullPointerException: freeDomainUrl must not be null

### ISSUE THREAD

**Issue by AliSoftware** [wordpress-mobile/WordPress-Android#15636]
**Sentry Url**: https://sentry.io/share/issue/268bfbfbc4ef4ec397b485c0e116624e/
**User Count**: 18
**Count**: 79
**First Release**: 18.7-rc-1
**First Seen**: 2021-11-15T17:02:34.411000Z
**Last Seen**: 2021-11-30T11:18:23Z
**24 Hours**: 13
**30 Days**: 78

```
NullPointerException: freeDomainUrl must not be null
     at org.wordpress.android.ui.domains.DomainsDashboardViewModel.buildDashboardItems(DomainsDashboardViewModel.kt:106)
     at org.wordpress.android.ui.domains.DomainsDashboardViewModel.access$onGetDomainClick(DomainsDashboardViewModel.kt:48)
     at org.wordpress.android.ui.domains.DomainsDashboardViewModel.access$buildDashboardItems(DomainsDashboardViewModel.kt:48)
     at org.wordpress.android.ui.domains.DomainsDashboardViewModel$refresh$1.invokeSuspend(DomainsDashboardViewModel.kt:96)
```

---

Quite low numbers so far, but since it's brand new in 18.7, which we just started to roll out to 10% yesterday, and given the trend according to the graphs, I seems it could potentially affect more and more users quickly as rollout progresses.

(And it being an NPE feels like it would somewhat also be an hopefully easy-ish enough to fix or at least workaround? Which is also why I filed it anyway)

**Comment by khaykov** [comment]
@renanferrari @ravishanker Looks like it might be related to the recent work on Domain Dashboard :)

**Comment by renanferrari** [comment]
Thanks for the ping @khaykov!

@ravishanker Let me know if you can handle this one, otherwise, I can take a look at it next week.

**Comment by jhnstn** [comment]
Hi @renanferrari and @ravishanker, seems like we already have an idea on the source of the issue here. Could one of you help set the priority so we can remove the triage status? Thanks, let me know if you need any help digging into the issue.

**Comment by renanferrari** [comment]
@jhnstn Thank you for the reminder, I'll take a look at this tomorrow.

**Comment by AliSoftware** [comment]
> [me in issue description, on Nov 30, 2021]:
> Quite low numbers so far, but […] it could potentially affect more and more users quickly as rollout progresses.

For refs/information/tracking purposes (and to help triage this): as of today, this is now at `5.7k events` / `1k users`.

**Comment by AliSoftware** [comment]
That being said, [it seems it's only reported for `18.7` and up to `18.8-rc-2`](https://sentry.io/organizations/a8c/issues/2792479856/tags/release/?project=1438088)… so maybe it was already fixed past `18.8-rc-2`?!

**Comment by renanferrari** [comment]
@AliSoftware You're right. I believe this was fixed by #15671.

**Comment by renanferrari** [comment]
Closed by #15671.

**Comment by antonis** [comment]
Linked the crash report above. The issue seems to persists in the latest version of the app or there is a regression. Pasting a stack trace from `23.3.1`:

```
java.lang.NullPointerException: freeDomainUrl must not be null
    at org.wordpress.android.ui.domains.DomainsDashboardViewModel.buildDashboardItems(DomainsDashboardViewModel.kt:106)
    at org.wordpress.android.ui.domains.DomainsDashboardViewModel.access$onGetDomainClick(DomainsDashboardViewModel.kt:45)
    at org.wordpress.android.ui.domains.DomainsDashboardViewModel.access$buildDashboardItems(DomainsDashboardViewModel.kt:45)
    at org.wordpress.android.ui.domains.DomainsDashboardViewModel$refresh$1.invokeSuspend(DomainsDashboardViewModel.kt:96)
    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
    at kotlinx.coroutines.CoroutineContextKt.withContinuationContext
    at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:90)
    at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
    at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
    at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
    at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
    at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
```
